stock_move_backdating/models/stock_move.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2015-2016 Agile Business Group (<http://www.agilebg.com>)
+# Copyright 2016 BREMSKERL-REIBBELAGWERKE EMMERLING GmbH & Co. KG
+#    Author Marco Dieckhoff
+# Copyright 2018 Alex Comba - Agile Business Group
+# Copyright 2023 Simone Rubino - TAKOBI
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+from odoo.fields import first
+from .stock_move_line import check_date
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _backdating_account_moves(self):
+        """Set date on linked account.move same for each move in `self`."""
+        picking_account_moves = self.env['account.move'].search(
+            [
+                ('stock_move_id', 'in', self.ids),
+            ],
+        )
+        for stock_move in self:
+            move_account_moves = picking_account_moves.filtered(
+                lambda am: am.stock_move_id == stock_move
+            )
+            move_account_moves.update({
+                'date': stock_move.date.date(),
+            })
+
+    def _backdating_action_done(self, moves_todo):
+        """Process the moves one by one, backdating the ones that need to."""
+        for move in self:
+            move_line = first(move.move_line_ids)
+            date_backdating = move_line.date_backdating
+            if date_backdating:
+                move = move.with_context(
+                    date_backdating=date_backdating,
+                )
+
+            moves_todo |= super(StockMove, move)._action_done()
```

Comment:
```suggestion
            moves_todo_ids = set()
            (...)

            move_todo_ids.add(super(StockMove, move)._action_done().ids)
            (...)
            
            return self.browse(moves_todo_ids)
```

Better for performances

Reply:
Thanks for having a look!
So this is for managing `ids` as much as possible, then `browse` the records only once: I guess this might be faster.
Do you have insight on _how much_ faster this is?

Reply:
I think we did the search few years ago. But I don't have the metrics anymore. Maybe @simahawk can help.

Reply:
@SirTakobi 

I've just wrote a little script :

```python

#!/usr/bin/env python
import logging

import click
import click_odoo

_logger = logging.getLogger(__name__)


@click.command()
@click_odoo.env_options(default_log_level='error')
def main(env):
    _initialize_records(env)
    _test_records(env)
    _test_ids(env)


def _initialize_records(env):
    values = list()
    for x in range(10000):
        values.append({
            "name": "Test %s" % str(x),
        })
    env["res.partner"].create(values)


def _test_records(env):
    _logger.info("BEGIN MERGE RECORDS")
    partners = env["res.partner"].browse()
    for partner in env["res.partner"].search([]):
        partners |= partner
    _logger.info("END MERGE RECORDS")


def _test_ids(env):
    _logger.info("BEGIN MERGE IDS")
    partners = set()
    for partner in env["res.partner"].search([]):
        partners.add(partner.id)
    env["res.partner"].browse(partners)
    _logger.info("END MERGE IDS")


if __name__ == "__main__":
    main()
```


And here is the output for the 10 000 records:

![image](https://github.com/OCA/stock-logistics-workflow/assets/19529533/521748e3-69d0-4980-bfeb-7e8dcbbbc153)


So, merging records take ~4 seconds and browsing ids 20 ms :smiley:

Reply:
And I've just 'base' module installed

Reply:
Very interesting, thanks!